plugins/inputs/kernel/kernel.go

```diff
@@ -98,31 +111,87 @@ func (k *Kernel) Gather(acc telegraf.Accumulator) error {
 		}
 	}
 
+	if k.optCollect["ksm"] {
+		stats := map[string]interface{}{
+			"full_scans":                         0,
+			"max_page_sharing":                   0,
+			"merge_across_nodes":                 0,
+			"pages_shared":                       0,
+			"pages_sharing":                      0,
+			"pages_to_scan":                      0,
+			"pages_unshared":                     0,
+			"pages_volatile":                     0,
+			"run":                                0,
+			"sleep_millisecs":                    0,
+			"stable_node_chains":                 0,
+			"stable_node_chains_prune_millisecs": 0,
+			"stable_node_dups":                   0,
+			"use_zero_pages":                     0,
+		}
+		// these exist in very recent Linux versions only, but useful to include if there.
+		extraStats := map[string]interface{}{
+			"general_profit": 0,
+		}
+
+		for f := range stats {
+			m, err := k.getProcValueInt(filepath.Join(k.ksmStatsDir, f))
+			if err != nil {
+				return err
+			}
+
+			fields["ksm_"+f] = m
+		}
+
+		for f := range extraStats {
+			m, err := k.getProcValueInt(filepath.Join(k.ksmStatsDir, f))
+			if err != nil {
+				// if an extraStats metric doesn't exist in our kernel version, ignore it.
+				continue
+			}
+
+			fields["ksm_"+f] = m
+		}
+	}
 	acc.AddCounter("kernel", fields, map[string]string{})
 
 	return nil
 }
 
-func (k *Kernel) getProcStat() ([]byte, error) {
-	if _, err := os.Stat(k.statFile); os.IsNotExist(err) {
-		return nil, fmt.Errorf("kernel: %s does not exist", k.statFile)
+func (k *Kernel) getProcValueBytes(path string) ([]byte, error) {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return nil, fmt.Errorf("kernel: Path %s does not exist", path)
 	} else if err != nil {
 		return nil, err
 	}
 
-	data, err := os.ReadFile(k.statFile)
+	data, err := os.ReadFile(path)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("kernel: Failed to read from %s", path)
```

Comment:
All error messages are already prefixed with the plugin name, so we could simply do
```suggestion
		return nil, fmt.Errorf("failed to read from %q: %w", path, err)
```

Reply:
ah didn't know that. Makes sense, will remove from all error messages and replace `%s` with `%q` while there.